01Matrix.js

```diff
@@ -0,0 +1,143 @@
+/*
+So we're going to represent our input as a grid where we're only concerned about the 1's, since we're trying 
+to find the distance for each 1 that is the closest to the nearest 0. 
+
+For this approach, we're going to use a multi-sourced BFS to find the distance to the nearest 0 
+and the way that works is we're going to gather the location of all the 0's first and add them to the queue,
+that way we can process them together. During the iteration, we're going to BFS each of those 
+locations in all 4 directions and until we reach a 1. Once we reach a 1, we mark it with the current 
+traveled distance so far.
+
+Since we're using a BFS, we can tranverse in levels and mark the locations we have visited and skip the ones
+we have already visited. That way can keep track of the distance traveled so far and only update it
+if we're visiting a location for the first time and since it will be the first time, it will be
+the shorest path.
+*/
+
+/* 
+Question:
+Multi-sourced BFS solution, passes on LeetCode 
+
+Given an m x n binary matrix mat, return the distance of the nearest 0 for each cell.
+The distance between two adjacent cells is 1.
+*/
+
+const { Queue } = require('../../../utils/queue');
+
+//    0  1  2
+// 0 [0, 0, 0],
+// 1 [0, 1, 0],
+// 2 [1, 1, 1],
+
+const directions = [
+    [-1, 0],
+    [0, -1],
+    [0, 1],
+    [1, 0],
+];
+
+const getRowColString = (rowCol) => {
+    const { row, col } = rowCol;
+
+    return `${row}, ${col}`;
+};
+
+const getZeroFilledMatrix = (grid, fillWithValue) => {
```

Comment:
This function is named in a misleading way. From the name, it sounds like this function always returns a matrix full of 0's. But, that's not what it does. You allow the user of this function to customize the fill value. So, your name should reflect this.